eth/chains/base.py

```diff
@@ -577,13 +585,23 @@ def import_block(self, block: BaseBlock, perform_validation: bool=True) -> BaseB
             ensure_imported_block_unchanged(imported_block, block)
             self.validate_block(imported_block)
 
-        self.chaindb.persist_block(imported_block)
+        new_canonical_headers, old_headers = self.chaindb.persist_block(imported_block)
         self.logger.debug(
             'IMPORTED_BLOCK: number %s | hash %s',
             imported_block.number,
             encode_hex(imported_block.hash),
         )
-        return imported_block
+
+        new_canonical_blocks = [
+            self.get_block_by_header(header) for header in new_canonical_headers]
+        old_canonical_blocks = [
+            self.get_block_by_header(header) for header in old_headers]
+
+        import_success = True
+        if not new_canonical_blocks:
+            new_canonical_blocks.append(imported_block)
+            import_success = False
+        return tuple(new_canonical_blocks), tuple(old_canonical_blocks), import_success
```

Comment:
I think I'd like to drop the `import_success` since it can be derived from the returned data.